### PR TITLE
Improve Waffle.py

### PR DESCRIPTION
**Pull request by Coal0**
Some things I changed:

* Wrapped long lines (146 characters! let's please take care of this when adding new arguments to `__init__`.);

* This entire file now uses double quotes consistently;

* Replaced string concatenation with `str.format` where it makes sense;

* Removed `enable` and `disable` methods (mixin?);

* Added whitespace around operators so they're easier to read and distinguish from things like starred expressions;

* Other minor PEP8 things;

* Implemented `__getitem__` so a `Waffle` object can pretend to be a 2D list. Unfortunately this one requires an extra class as far as I can see.

### CODE REVIEW

File: guizero/Waffle.py [hunk 1]
```diff
@@ -106,34 +168,22 @@ def get_all(self):
...
-                    self._command(pixel_x,pixel_y)
-                else:
-                    utils.error_format("Waffle command function must accept either 0 or 2 arguments.\nThe current command has {} arguments.".format(args_expected))
-
-    def update_command(self, command):
-        if command is None:
-            self._command = lambda: None
-        else:
-            self._command = command
+                self._command(pixel_x, pixel_y)
```
Comment (martinohanlon) [inline guizero/Waffle.py]: you should be using ._update_command here which you seem to have removed? perhaps your branch was out of date?

File: guizero/Waffle.py [hunk 2]
```diff
@@ -142,14 +192,6 @@ def enabled(self):
     @enabled.setter
     def enabled(self, value):
         self._enabled = value
-    
-    def disable(self):
```
Comment (martinohanlon) [inline guizero/Waffle.py]: enabled \ disabled should be implemented in Waffle as there implementation is specific to Waffle and not shared with any other component.

File: guizero/Waffle.py [hunk 3]
```diff
@@ -106,34 +168,22 @@ def get_all(self):
...
             canvas = e.widget
             x = canvas.canvasx(e.x)
             y = canvas.canvasy(e.y)
-            pixel_x = int(x/(self._pixel_size+self._pad))
-            pixel_y = int(y/(self._pixel_size+self._pad))
+            pixel_x = int(x / (self._pixel_size + self._pad))
+            pixel_y = int(y / (self._pixel_size + self._pad))
             if self._command:
-                args_expected = utils.no_args_expected(self._command)
-                if args_expected == 0:
```
Comment (martinohanlon) [inline guizero/Waffle.py]: you have removed the code which caters for either 0 or 2 arguments being passed.

File: guizero/Waffle.py [hunk 4]
```diff
@@ -20,9 +66,9 @@ def __init__(self, master, height=3, width=3, dim=20, pad=5, color="white", dott
         self._enabled = True
 
     	# Description of this object (for friendly error messages)
-        self.description = "[Waffle] object ("+str(height)+"x"+str(width)+")"
+        self.description = "[Waffle] object ({}x{})".format(height, width)
 
-        self.update_command(command)
+        self._command = command
         self._height = height       # How many pixels high
```
Comment (martinohanlon) [inline guizero/Waffle.py]: you should be using ._update_command here which you seem to have removed? perhaps your branch was out of date?

**Comment by martinohanlon** [comment]
note... docs will need updating.

**Comment by Coal0** [comment]
Sorry for the belated response- I didn't get to work on this yet, but I'll see to it soon.

**Comment by martinohanlon** [comment]
Thanks a lot, I have refactored Waffle to resolve #125 #126 and used your PR as a cross reference and help.

**Pull request closed without merging**